src/savedata.c

```diff
@@ -579,274 +578,269 @@ static void SaveDataExtra_LoadCheck (SaveData *saveData, int *frontierResult, in
     }
 
     SleepUnlock(1);
 }
 
-static void sub_020250DC (SaveData * param0, UnkStruct_020250DC * param1)
+static void SaveDataState_Cancel (SaveData *saveData, SaveDataState *saveState)
 {
-    int v0;
+    int i;
 
-    if (param1->unk_00) {
-        param0->globalCounter = param1->unk_18;
+    if (saveState->fullSaveMode) {
+        saveData->globalCounter = saveState->globalCounterBackup;
     }
 
-    for (v0 = 0; v0 < 2; v0++) {
-        param0->unk_20018[v0] = param1->unk_1C[v0];
+    for (i = 0; i < SAVE_BLOCK_ID_MAX; i++) {
+        saveData->blockCounters[i] = saveState->blockCounterBackup[i];
     }
 
     if (!CARD_TryWaitBackupAsync()) {
         CARD_CancelBackupAsync();
     }
 
-    if (param1->unk_24) {
-        CARD_UnlockBackup(param1->unk_10);
-        OS_ReleaseLockID(param1->unk_10);
-        param1->unk_24 = 0;
+    if (saveState->isLocked) {
+        CARD_UnlockBackup(saveState->lockID);
+        OS_ReleaseLockID(saveState->lockID);
+        saveState->isLocked = FALSE;
     }
 
     SleepUnlock(1);
 }
 
-BOOL sub_0202513C (SaveData * param0)
+BOOL SaveDataState_Save (SaveData *saveData)
 {
-    int v0;
-    UnkStruct_020250DC v1;
+    int saveResult;
+    SaveDataState saveState;
 
-    sub_02024EC8(param0, &v1, 2);
+    SaveDataState_Init(saveData, &saveState, SAVE_BLOCK_ID_MAX);
 
     do {
-        v0 = sub_02024F44(param0, &v1);
-    } while (v0 == 0 || v0 == 1);
+        saveResult = SaveDataState_Main(saveData, &saveState);
+    } while (saveResult == SAVE_RESULT_PROCEED || saveResult == SAVE_RESULT_PROCEED_FINAL);
 
-    sub_0202506C(param0, &v1, v0);
-    return v0;
+    SaveDataState_End(saveData, &saveState, saveResult);
+    return saveResult;
 }
 
-static BOOL sub_0202516C (const SaveData * param0, int param1, int param2)
+static BOOL SaveBlockFooter_Erase (const SaveData *saveData, int blockID, int sectorID)
 {
-    u32 v0;
-    SaveBlockFooter v1;
-    const SaveBlockInfo * v2 = &param0->blockInfo[param1];
+    u32 saveOffset;
+    SaveBlockFooter footer;
+    const SaveBlockInfo* blockInfo = &saveData->blockInfo[blockID];
 
-    MI_CpuFill8(&v1, 0xff, sizeof(SaveBlockFooter));
-    v0 = sub_02024888(param2, v2) + v2->size - sizeof(SaveBlockFooter);
+    MI_CpuFill8(&footer, 0xff, sizeof(SaveBlockFooter));
+    saveOffset = SaveData_SaveOffset(sectorID, blockInfo) + blockInfo->size - sizeof(SaveBlockFooter);
 
-    return sub_02025A9C(v0, &v1, sizeof(SaveBlockFooter));
+    return sub_02025A9C(saveOffset, &footer, sizeof(SaveBlockFooter));
 }
 
-int sub_020251A4 (int param0)
+int SaveTableEntry_BodySize (int saveTableID)
 {
-    int v0;
-    const SaveTableEntry * v1 = gSaveTable;
+    int size;
+    const SaveTableEntry* saveTable = gSaveTable;
 
-    GF_ASSERT(param0 < gSaveTableSize);
-    v0 = v1[param0].sizeFunc();
+    GF_ASSERT(saveTableID < gSaveTableSize);
+    size = saveTable[saveTableID].sizeFunc();
 
-    v0 += 4 - (v0 % 4);
-    v0 += 4;
+    size += 4 - (size % 4);
+    size += 4;
```

Comment:
Is there a potential const for this 4?

Reply:
From what I can tell it looks like padding, so the significance just seems to be that it's half a byte. I'll plan on leaving as-is unless you disagree.

Reply:
Fine by me.